typescript-playwright-sample/azure-pipelines.yml

```diff
@@ -0,0 +1,97 @@
+# documentation at https://docs.microsoft.com/en-us/azure/devops/pipelines/create-first-pipeline
+# (using this sample as your starter code).
+#
+# For more information, see:
+# * https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started
+# * https://docs.microsoft.com/en-us/azure/devops/pipelines/customize-pipeline
+
+# The browser automation tool we're using, Playwright, works with many different browsers. Here,
+# we show an example of using the same test code with different combinations of browsers and OSes.
+#
+# For more information, see:
+# * https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started-multiplatform
+# * https://playwright.dev/docs/test-intro
+strategy:
+  matrix:
+    win-chromium:
+      imageName: windows-2019
+      browser: chromium
+    linux-firefox:
+      imageName: ubuntu-20.04
+      browser: firefox
+    mac-webkit:
+      imageName: macOS-10.15
+      browser: webkit
+
+pool:
+  vmImage: $(imageName)
+
+steps:
+  # This task makes sure Node.js is installed on the build agent.
+  #
+  # It is a good idea to test against a specific, known version of Node; this makes it easier for other users
+  # to reproduce your build results.
+  - task: NodeTool@0
+    displayName: install node 14.17.0
+    inputs:
+      versionSpec: "14.17.0"
+
+  # This task installs dependencies specified in your package.json file.
+  #
+  # Our sample uses Yarn for this, but if your project uses NPM instead, that's also fine.
+  - script: yarn install --frozen-lockfile # or "npm install --ci"
+    displayName: yarn install
+    # workingDirectory should match where your package.json file is located. It defaults to the root of the repository.
+    workingDirectory: $(Build.SourcesDirectory)/typescript-playwright-sample
+
+  # This task runs the tests specified by playwright.config.js, including our example accessibility tests.
+  #
+  # If your project already uses npm instead of yarn, or if your project already runs "playwright test" directly here,
+  # those are both fine.
+  - script: yarn test --browser=$(browser) # or "npm run test -- --browser=$(browser)"
+    displayName: yarn test
+    workingDirectory: $(Build.SourcesDirectory)/typescript-playwright-sample
+
+  # This task is how the "Tests" tab in our Azure Pipelines build results page gets its data
+  - task: PublishTestResults@2
+    displayName: publish test results
+    inputs:
+      testResultsFiles: $(Build.SourcesDirectory)/typescript-playwright-sample/test-results/junit.xml
+      testRunTitle: typescript-playwright-sample ($(imageName) $(browser))
+    condition: succeededOrFailed()
+
+  # If you are adding tests to an existing project that already uses Azure Pipelines, this is probably the only
+  # new task you'll need to add.
+  #
+  # This task is how the "Scans" tab in our Azure Pipelines build results page gets its data
+  - task: PublishBuildArtifacts@1
```

Comment:
Some pipelines might have `PublishPipelineArtifacts` now because the `PublishBuildArtifacts` docs have a [deprecation](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-build-artifacts?view=azure-devops) notice. However, the SARIF task requires the older `PublishBuildArtifacts` - mentioned in [the Q&A](https://marketplace.visualstudio.com/items?itemName=sariftools.scans&ssr=false#qna) (I also ran into this same problem the other day when exploring SARIF for the ADO task prototype). I'm not sure if it's worth adding a sentence here about it

Reply:
Update the comment to clarify that using the old task is intentional/required